src/app/my-skills/my-skills-dialog-component-trait.ts

```diff
@@ -0,0 +1,27 @@
+import { MatSlider } from '@angular/material';
+import { StepDescription } from './step-description';
+
+export abstract class MySkillsDialogComponentTrait {
+
+  protected updateLabel(slider: MatSlider, description: StepDescription, step: number) {
+    if (step === null) {
+      return;
+    }
+
+    const title = description['step' + step];
+    slider._elementRef.nativeElement.querySelector('.mat-slider-thumb').setAttribute('title', title);
```

Comment:
Could this behavior of updating the title of the slider depending on its current value be implemented as an Angular directive? The the `StepDescription` object would still be loaded outside by the component and passed as a parameter to the directive. What do you think?

Reply:
Thank you very much for the suggestion :blush: